enums.rb

```diff
@@ -0,0 +1,136 @@
+# rubocop: disable Metrics/ModuleLength, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+# Description/Explanation of module enums
+module Enumerable
+  #:nodoc:
+  def my_each(index = 0)
+    return to_enum unless block_given?
+
+    while index < to_a.length
+      if instance_of?(Hash)
+        yield(keys[index], values[index])
+      else
+        yield(to_a[index])
+      end
+      index += 1
+    end
+    self
+  end
+
+  def my_each_with_index(index = 0)
+    return to_enum unless block_given?
+
+    while index < to_a.length
+      yield(to_a[index], index)
+      index += 1
+    end
+    self
+  end
+
+  def my_select(arr = [], index = 0)
+    return to_enum unless block_given?
+
+    my_each do |k|
+      if yield(k)
+        arr[i] = k
+        index += 1
+      end
+    end
+    arr
+  end
```

Comment:
- [ ]  returns an array containing all elements of enum for which the given block returns a true value

```
  array = [1,2,3,4,5,6]
  block = proc { |num| num < (0 + 9) / 2 }
  p array.select(&block) # [1, 2, 3]
  p array.my_select(&block) # Error
```

- [ ] does not mutate the original array

```
  array.my_select { |num| num + 1 } # Error
```

Reply:
E: Lint/Syntax: unexpected token kEND
(Using Ruby 2.4 parser; configure using TargetRubyVersion parameter, under AllCops)
  end
can you telll why this error is showing?